pkg/registry/networking/networkpolicy/strategy_test.go

```diff
@@ -0,0 +1,235 @@
+			hasEndPort:        true,
+			enableFeatureGate: false,
+			isIngress:         false,
+			isEgress:          true,
+		},
+		{
+			name:              "Create Egress Rule with EndPort Feature Gate true and with endPort undefined",
+			hasEndPort:        false,
+			enableFeatureGate: true,
+			isIngress:         false,
+			isEgress:          true,
+		},
+		{
+			name:              "Create Egress Rule with EndPort Feature Gate disabled and with endPort undefined",
+			hasEndPort:        false,
+			enableFeatureGate: false,
+			isIngress:         false,
+			isEgress:          true,
+		},
+		{
+			name:              "Create Ingress and Egress Rule with EndPort Feature Gate enabled and endPort defined",
+			hasEndPort:        true,
+			enableFeatureGate: true,
+			isIngress:         true,
+			isEgress:          true,
+		},
+		{
+			name:              "Create Ingress and Egress Rule with EndPort Feature Gate disabled and endPort defined",
+			hasEndPort:        true,
+			enableFeatureGate: false,
+			isIngress:         true,
+			isEgress:          true,
+		},
+		{
+			name:              "Create Ingress and Egress Rule with EndPort Feature Gate enabled and endPort undefined",
+			hasEndPort:        false,
+			enableFeatureGate: true,
+			isIngress:         true,
+			isEgress:          true,
+		},
+		{
+			name:              "Create Ingress and Egress Rule with EndPort Feature Gate disabled and endPort undefined",
+			hasEndPort:        false,
+			enableFeatureGate: false,
+			isIngress:         true,
+			isEgress:          true,
+		},
+		{
+			name:              "Create a null rule with EndPort Feature Gate enabled",
+			hasEndPort:        false,
+			enableFeatureGate: true,
+			isIngress:         false,
+			isEgress:          false,
+		},
+		{
+			name:              "Create a null rule with EndPort Feature Gate disabled",
+			hasEndPort:        false,
+			enableFeatureGate: false,
+			isIngress:         false,
+			isEgress:          false,
+		},
+	} {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.NetworkPolicyEndPort, tc.enableFeatureGate)()
+		netpol := makeNetworkPolicy(tc.isIngress, tc.isEgress, tc.hasEndPort)
+		Strategy.PrepareForCreate(context.Background(), netpol)
+		err := ValidateNetworkPolicy(netpol, tc.enableFeatureGate, tc.hasEndPort)
+		if err != nil {
+			t.Errorf("Create Rule test %s failed: %s", tc.name, err)
+		}
+
+		oldNetPol := makeNetworkPolicy(tc.isIngress, tc.isEgress, false)
+		newNetPol := makeNetworkPolicy(tc.isIngress, tc.isEgress, tc.hasEndPort)
+		Strategy.PrepareForUpdate(context.Background(), newNetPol, oldNetPol)
+		err = ValidateNetworkPolicy(newNetPol, tc.enableFeatureGate, tc.hasEndPort)
+		if err != nil {
+			t.Errorf("Update Rule test %s failed: %s", tc.name, err)
+		}
+
+	}
+}
+
+func ValidateNetworkPolicy(netPol *networking.NetworkPolicy, fgEnabled, hasEndPort bool) error {
```

Comment:
Why is this in the test file? :P

Reply:
fwiw we shouldn't have to have the validate logic tested in the rest strategy code

Reply:
Yeap, so my approach here was to, instead of repeat the code, create a function just to verify not the validation itself, but if the field has been properly dropped (or if it has been dropped and shouldn't).

Reply:
Ah I see, this is testing whether the network policy is valid, I thought it was the validation code.

Reply:
Could we instead include a "before" and "after" NetworkPolicy in the test table and do a `DeepEqual` against the expected output?

Reply:
maybe im wrong.. but does this need to be capitalized if it is used only in this pkg? 
```suggestion
func validateNetworkPolicy(netPol *networking.NetworkPolicy, fgEnabled, hasEndPort bool) error {
```

Reply:
@andrewsykim ok, will do. Tests are going to be HUGE but :man_shrugging: as we spoke in Slack, 60 LOC, 700 Lines of tests, "this is the way" :P

Reply:
> maybe im wrong.. but does this need to be capitalized if it is used only in this pkg?

Yeap, you're right. Will un-export the function.